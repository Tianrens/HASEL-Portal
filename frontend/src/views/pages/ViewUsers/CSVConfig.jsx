import dayjs from 'dayjs';

export const csvOptions = {
    // Order of headers
    headers: [
        'firstName',
        'lastName',
        'upi',
        'email',
        'type',
        'currentRequestId.allocatedWorkstationId.name',
        'currentRequestId.status',
        'currentRequestId.startDate',
        'currentRequestId.endDate',
    ],
    // Rename headers to something more readible
    rename: [
        'First Name',
        'Last Name',
        'UPI',
        'Email',
        'Account Type',
        'Allocated Workstation',
        'Request Status',
        'Request Start Date',
        'Request Expiry Date',
        ' ', // Remove headers for id fields, cannot delete so have to rename to blank
        ' ',
        ' ',
        ' ',
    ],
    // This lets us modify values
    typeHandlers: {
        String: (value, index) => {
            // Output date to a readible format
            if (dayjs(value).isValid()) {
                return dayjs(value).format('ddd DD/MMM/YYYY');
            }
            // Remove any id fields
            if (index === 'authUserId' || index === '_id') {
                return '';
            }
            return value;
        },
    },
};
