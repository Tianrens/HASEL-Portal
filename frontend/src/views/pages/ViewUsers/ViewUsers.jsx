import { React } from 'react';
import { TableCell, TableRow } from '@mui/material';
import { Link } from 'react-router-dom';
import TopBarPageTemplate from '../../components/templates/TopBarPageTemplate/TopBarPageTemplate';
import StyledHeader from '../../components/text/StyledHeader';
import TableWithPagination from '../../components/table/TableWithPagination';
import { getDisplayName } from '../../../config/accountTypes';
import styles from './ViewUsers.module.scss';

const columns = ['Name', 'UPI', 'Type', 'Status', 'Date Approved'];

const rowFactory = (user) => (
    <TableRow key={user._id} component={Link} to={`/users/${user._id}`}>
        <TableCell component='th' scope='row'>
            {user.firstName} {user.lastName}
        </TableCell>
        <TableCell align='right'>{user.upi}</TableCell>
        <TableCell align='right'>{getDisplayName(user.type)}</TableCell>
        <TableCell align='right'>{user?.currentRequestId?.status ?? '-'}</TableCell>
        <TableCell align='right'>
            {user?.currentRequestId?.startDate
                ? new Date(user.currentRequestId.startDate).toLocaleDateString()
                : '-'}
        </TableCell>
    </TableRow>
);

const ViewUsers = () => (
    <TopBarPageTemplate>
        <div className={styles.container}>
            <StyledHeader left>Users</StyledHeader>
            <TableWithPagination
                endpoint='/api/user'
                rowProp='users'
                columns={columns}
                rowFactory={rowFactory}
            />
        </div>
    </TopBarPageTemplate>
);

export default ViewUsers;
