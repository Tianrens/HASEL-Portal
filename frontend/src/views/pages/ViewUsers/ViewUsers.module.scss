.searchIndicator {
    height: 1em;
    margin-top: 0;
}

.hidden {
    display: none;
}

.italic {
    font-weight: normal;
}
