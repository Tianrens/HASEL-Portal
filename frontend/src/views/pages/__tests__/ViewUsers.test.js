import React from 'react';
import renderer from 'react-test-renderer';
import StylesProvider from '@mui/styles/StylesProvider';
import { MemoryRouter } from 'react-router-dom';
import ViewUsers from '../ViewUsers/ViewUsers';

test('View Users page renders properly', () => {
    const snapshotComponent = renderer.create(
        <StylesProvider injectFirst>
            <MemoryRouter>
                <ViewUsers />
            </MemoryRouter>
        </StylesProvider>,
    );
    const tree = snapshotComponent.toJSON();
    expect(tree).toMatchSnapshot();
});
