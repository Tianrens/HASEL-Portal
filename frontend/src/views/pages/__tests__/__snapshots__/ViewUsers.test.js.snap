// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Users page renders properly 1`] = `
Array [
  <div
    className="fillTopbarSpace"
  >
    <div
      className="navbar"
    >
      <div
        className="patternBgLayer"
      >
        <div
          className="bgLayer"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="University of Auckland"
              className="uoaLogo"
              src="uoaLogo.svg"
            />
            <div
              className="spacer"
            />
            <div
              className="title"
            >
              HASEL Lab
            </div>
          </a>
          <div
            className="pushSpacer"
          />
          <div
            className="horizontalNavLinks"
          >
            <div
              className="linksWrapper"
            >
              <button
                className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root wrapper css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                disabled={false}
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <a
                  href="/user"
                  onClick={[Function]}
                >
                  <img
                    alt="My Account"
                    className="navIcon"
                    src="id.svg"
                  />
                  <div
                    className="spacer"
                  />
                  <div
                    className="title"
                  >
                    My Account
                  </div>
                </a>
              </button>
            </div>
          </div>
          <div
            className="drawerWrapper"
            onClick={[Function]}
            onTouchEnd={[Function]}
          >
            <button
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge drawerButton css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DehazeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M2 15.5v2h20v-2H2zm0-5v2h20v-2H2zm0-5v2h20v-2H2z"
                />
              </svg>
            </button>
            <div
              className="MuiDrawer-root MuiDrawer-docked drawer css-7ik032-MuiDrawer-docked"
            >
              <div
                className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight css-18sg6k4-MuiPaper-root-MuiDrawer-paper"
                style={
                  Object {
                    "visibility": "hidden",
                  }
                }
              >
                <div
                  className="drawerPatternBg"
                >
                  <div
                    className="drawer"
                  >
                    <div>
                      <button
                        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge drawerButton css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                        disabled={false}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onContextMenu={[Function]}
                        onDragLeave={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        tabIndex={0}
                        type="button"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CloseIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                          />
                        </svg>
                      </button>
                    </div>
                    <div
                      className="linksWrapper"
                    >
                      <button
                        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root wrapper css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                        disabled={false}
                        onBlur={[Function]}
                        onContextMenu={[Function]}
                        onDragLeave={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        tabIndex={0}
                        type="button"
                      >
                        <a
                          href="/user"
                          onClick={[Function]}
                        >
                          <img
                            alt="My Account"
                            className="navIcon"
                            src="id.svg"
                          />
                          <div
                            className="spacer"
                          />
                          <div
                            className="title"
                          >
                            My Account
                          </div>
                        </a>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="content"
  >
    <div
      className="container"
    >
      <div
        className="container"
      >
        <h2
          className="text alignLeft "
        >
          <span>
            Users
          </span>
        </h2>
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
      >
        <table
          className="MuiTable-root table css-rqglhn-MuiTable-root"
          role={null}
        >
          <thead
            className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              role={null}
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium headerCell css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Name
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium headerCell css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                UPI
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium headerCell css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Type
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium headerCell css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Status
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium headerCell css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Date Approved
              </th>
            </tr>
          </thead>
          <tbody
            className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            role={null}
          />
        </table>
      </div>
      <div
        className="MuiTablePagination-root css-jtlhu6-MuiTablePagination-root"
      >
        <div
          className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-11bfvty-MuiToolbar-root-MuiTablePagination-toolbar"
        >
          <div
            className="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
          />
          <p
            className="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
          >
            0-0 of 0
          </p>
          <div
            className="MuiTablePagination-actions"
          >
            <button
              aria-label="Go to previous page"
              className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Go to previous page"
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowLeftIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </button>
            <button
              aria-label="Go to next page"
              className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Go to next page"
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowRightIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
